src/Components/JoinForm/JoinForm.js

```diff
@@ -2,13 +2,17 @@ import React, { useState, useEffect } from 'react';
 import './JoinForm.css';
 import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
+import { addNewCode } from './../../Actions';
 
-function JoinForm({ codes }) {
+function JoinForm({ codes, addNewCode }) {
   const [ nickname, setNickname ] = useState('')
   const [ roomCode, setRoomCode ] = useState('')
 
   const saveMyCodes = () => {
-    window.localStorage.setItem('codes', JSON.stringify( codes ))
+    addNewCode(roomCode)
+    if ( !codes.includes(roomCode)) {
+      window.localStorage.setItem('codes', JSON.stringify( [roomCode, ...codes] ))
```

Comment:
this is what allows us to add entered codes into the local storage?

Reply:
Yeah, this checks that it's a new code that local storage doesn't have and then overwrites local with the updated list.